packages/gatsby-recipes/src/providers/gatsby/plugin.js

```diff
@@ -243,7 +264,7 @@ const read = async ({ root }, id) => {
       plugin => plugin.key === id || plugin.name === id
     )
 
-    if (plugin) {
+    if (plugin?.name) {
```

Comment:
Isn't this backwards breaking in some way?

Reply:
I added this because in the "non-supported" way it would break on this instead of going till the explicit error.

Reply:
So before this would never work? Because `plugin?.name` is not the same as `plugin?.name || plugin` so that's why it feels like a breaking change to me. I can't really tell whether or not the rest of this PR is mandating this change, or that it's an external plugin api change, or just didn't work before.

Reply:
No, this wouldn't work at all. When the config is not supported it can't find any plugins and thus returned nothing for `plugin.name` and then it broke on the function below of that.

Reply:
Good fix then :)